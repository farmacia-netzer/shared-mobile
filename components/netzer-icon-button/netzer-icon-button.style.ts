import { StyleSheet } from 'react-native';
import { GRAY_SCALE } from '##theme/grayscale.constant';
import { MEDIUM_PADDING, NORMAL_MARGIN } from '##theme/dimensions.constant';

export const FAB_DIMENSIONS = 50;
const FAB_BORDER_RADIUS = FAB_DIMENSIONS / 2;
const MINI_FAB_DIMENSIONS = 32;
const MINI_FAB_BORDER_RADIUS = MINI_FAB_DIMENSIONS / 2;

const MINI_FAB_EXT_IMG_HEIGHT = 12;
const MINI_FAB_EXT_HEIGHT = MINI_FAB_DIMENSIONS;
const MINI_FAB_EXT_BORDER_RADIUS = MINI_FAB_EXT_HEIGHT / 2;
const OUTLINE_ICON_DIAMETER = 50;
export const DEFAULT_ICON_COLOR = GRAY_SCALE.GRAY_20;

const commonStyles = StyleSheet.create({
  FABShadow: {
    backgroundColor: GRAY_SCALE.WHITE,
    shadowColor: GRAY_SCALE.BLACK,
    shadowOffset: { width: 0, height: 0 },
    shadowOpacity: 0.2,
    shadowRadius: 4,
    elevation: 4
  }
});

export const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row'
  },
  contentContainer: {
    flexDirection: 'column',
    alignItems: 'center'
  },
  defaultIconStyle: {
    justifyContent: 'center'
  },
  defaultTextStyle: {
    color: DEFAULT_ICON_COLOR,
    justifyContent: 'center',
    textAlign: 'center'
  },
  iconOutline: {
    borderWidth: 2,
    borderColor: DEFAULT_ICON_COLOR,
    height: OUTLINE_ICON_DIAMETER,
    width: OUTLINE_ICON_DIAMETER,
    borderRadius: OUTLINE_ICON_DIAMETER / 2,
    alignItems: 'center',
    justifyContent: 'center'
  },
  outlineText: {
    marginTop: NORMAL_MARGIN,
    color: DEFAULT_ICON_COLOR
  },
  FAB: {
    width: FAB_DIMENSIONS,
    height: FAB_DIMENSIONS,
    borderRadius: FAB_BORDER_RADIUS,
    ...commonStyles.FABShadow
  },
  miniFAB: {
    width: MINI_FAB_DIMENSIONS,
    height: MINI_FAB_DIMENSIONS,
    borderRadius: MINI_FAB_BORDER_RADIUS,
    ...commonStyles.FABShadow
  },
  miniFABext: {
    borderRadius: MINI_FAB_EXT_BORDER_RADIUS,
    paddingHorizontal: MEDIUM_PADDING,
    height: MINI_FAB_EXT_HEIGHT,
    ...commonStyles.FABShadow
  },
  miniFABextImage: {
    height: MINI_FAB_EXT_IMG_HEIGHT,
    resizeMode: 'contain'
  }
});
