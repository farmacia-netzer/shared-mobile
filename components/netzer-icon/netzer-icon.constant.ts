export const DEFAULT_ICON_SIZE = 22;

export interface GLYPH {
  family: string;
  name: string;
}
export interface ICON_GLYPH {
  [k: string]: GLYPH;
}
export const ICON_GLYPH_MAP: ICON_GLYPH = {
  CHEVRON_LEFT: { family: 'netzer', name: 'netzer-chevron-left' },

  ADD_SHOPPING_CART: { family: 'netzer', name: 'netzer-add-to-cart' },
  ADD_SHOPPING_LIST: { family: 'netzer', name: 'netzer-list-add' },
  ADD_CONTACT_INFO: { family: 'netzer', name: 'netzer-add-contact-info' },
  ADD_TIMESLOT: { family: 'netzer', name: 'netzer-select-timeslot' },
  AGE_RESTRICTED: { family: 'netzer', name: 'netzer-identification' },
  BARCODE_SCAN: { family: 'netzer', name: 'netzer-barcode-scan' },
  CALENDAR: { family: 'netzer', name: 'netzer-calendar' },
  CHEVRON_RIGHT: { family: 'netzer', name: 'netzer-chevron-right' },
  CLEAR: { family: 'netzer', name: 'netzer-close' },
  CLIP: { family: 'netzer', name: 'netzer-cut' },
  COUPON_CLIP: { family: 'netzer', name: 'netzer-coupon' },
  CLOCK: { family: 'fontawesome', name: 'clock' },
  CLOSE: { family: 'netzer', name: 'netzer-close-1' },
  EDIT_PENCIL: { family: 'netzer', name: 'netzer-edit-pencil' },
  FORBIDDEN: { family: 'netzer', name: 'netzer-forbidden' },
  MENU: { family: 'netzer', name: 'netzer-hamburger' },
  MISSING_PAGE: { family: 'netzer', name: 'netzer-missing-page' },
  PIG_BANK: { family: 'netzer', name: 'netzer-piggybank' },
  LOCATION: { family: 'netzer', name: 'netzer-location-pin' },
  LOYALTY_POINTS_CARD: { family: 'netzer', name: 'netzer-loyalty-points-card' },
  REWARDS: { family: 'netzer', name: 'netzer-rewards' },
  SEARCH: { family: 'netzer', name: 'netzer-search' },
  SHOP: { family: 'netzer', name: 'netzer-shop' },
  SMARTPHONE: { family: 'netzer', name: 'netzer-smartphone' },
  SNAP: { family: 'netzer', name: 'netzer-snap' },
  SNAP_EBT: { family: 'netzer', name: 'netzer-snap-ebt' },
  SORT: { family: 'netzer', name: 'netzer-sort' },
  STAR: { family: 'netzer', name: 'netzer-star' },
  PLUS: { family: 'netzer', name: 'netzer-plus' },
  MINUS: { family: 'fontawesome', name: 'minus' },
  LOGIN: { family: 'netzer', name: 'netzer-avatar' },
  YOUTUBE: { family: 'netzer', name: 'netzer-youtube' },
  CHECK: { family: 'netzer', name: 'netzer-checkmark' },
  CHECK_CIRCLE: { family: 'fontawesomeicon', name: 'check-circle' },
  REFINE: { family: 'netzer', name: 'netzer-filter' },
  VIEW: { family: 'netzer', name: 'netzer-view' },
  DIRECTIONS: { family: 'netzer', name: 'netzer-directions' },
  STORE_FLYERS: { family: 'netzer', name: 'netzer-store-flyer' },
  STOREFRONT: { family: 'netzer', name: 'netzer-store' },
  CALL: { family: 'netzer', name: 'netzer-call' },
  TRIANGLE_DOWN: { family: 'netzer', name: 'netzer-triangle-down' },
  TIMER: { family: 'netzer', name: 'netzer-inprogress' },
  IMAGE: { family: 'materialcommunityicons', name: 'image' },
  REFRESH: { family: 'fontawesomeicon', name: 'refresh' },
  EYE: { family: 'fontawesomeicon', name: 'eye' },
  EYE_SLASH: { family: 'fontawesomeicon', name: 'eye-slash' },
  OUTLINED_CIRCLE: { family: 'fontawesomeicon', name: 'circle-thin' },
  CIRCLE: { family: 'netzer', name: 'netzer-close-round' },
  MAP_MARKER: { family: 'fontawesome', name: 'map-marker' },
  EXTERNAL_LINK: { family: 'fontawesome', name: 'external-link-alt' },
  LOCATE: { family: 'materialicons', name: 'my-location' },
  TRASH_CAN: { family: 'netzer', name: 'netzer-trash' },
  OK_CIRCLED: { family: 'netzer', name: 'netzer-ok-circled' },
  CANCEL_CIRCLED: { family: 'netzer', name: 'netzer-cancel-circled' },
  CHEVRON_UP: { family: 'netzer', name: 'netzer-chevron-up' },
  CHEVRON_THIN_UP: { family: 'netzer', name: 'netzer-thin-chevron-up' },
  SPINNER: { family: 'fontawesomeicon', name: 'spinner' },
  ALERT_CIRCLE: { family: 'netzer', name: 'netzer-alert-circle' },
  EXCLAMATION_CIRCLE: { family: 'fontawesome', name: 'exclamation-circle' },
  DELIVERY_TRUCK: { family: 'netzer', name: 'netzer-delivery-truck' },
  SHOP_BAG: { family: 'netzer', name: 'netzer-shop-bag' },
  NAVIGATION: { family: 'netzer', name: 'netzer-navigation' },
  ARROW_UP_LEFT: { family: 'netzer', name: 'netzer-arrow-up-left' },
  CC_VISA: { family: 'fontawesomeicon', name: 'cc-visa' },
  CC_DINERS: { family: 'fontawesomeicon', name: 'cc-diners-club' },
  CC_MASTERCARD: { family: 'fontawesomeicon', name: 'cc-mastercard' },
  CC_JCB: { family: 'fontawesomeicon', name: 'cc-jcb' },
  CC_AMEX: { family: 'fontawesomeicon', name: 'cc-amex' },
  CC_DISCOVER: { family: 'fontawesomeicon', name: 'cc-discover' },
  CART_PICKUP: { family: 'netzer', name: 'netzer-pickup' },
  INFO: { family: 'materialcommunityicons', name: 'information' },
  RADIO_BOX_UNSELECTED: { family: 'materialcommunityicons', name: 'radiobox-blank' },
  RADIO_BOX_SELECTED: { family: 'materialcommunityicons', name: 'radiobox-marked' },
  PARITY: { family: 'fontawesomeicon', name: 'table' },
  ADD_PAYMENT_CARD: { family: 'netzer', name: 'netzer-add-payment-card' },
  NOTES: { family: 'netzer', name: 'netzer-notes' },
  ADD_TO_LIST: { family: 'netzer', name: 'netzer-add-to-list' },
  SHARE_IOS: { family: 'netzer', name: 'netzer-share-ios' },
  SHARE_ANDROID: { family: 'netzer', name: 'netzer-share-android' },
  DEAL_GROUP: { family: 'netzer', name: 'netzer-deal-group' },

  // ionicons
  HOME: { family: 'ionicons', name: 'home-outline' },
  PRODUCTS: { family: 'ionicons', name: 'color-filter-outline' },
  CART: { family: 'ionicons', name: 'cart' },
  PROFILE: { family: 'ionicons', name: 'person-outline' },
  ELLIPSIS_H: { family: 'ionicons', name: 'ios-ellipsis-horizontal' },
  TRASH_OUTLINE: { family: 'ionicons', name: 'trash-outline' },
  CART_OUTLINE: { family: 'ionicons', name: 'cart-outline' },
  REMOVE_OUTLINE: { family: 'ionicons', name: 'remove-outline' },
  ADD_OUTLINE: { family: 'ionicons', name: 'add-outline' },
  VERTICAL_ELLIPSIS: { family: 'ionicons', name: 'ios-ellipsis-vertical' },
  CHECKBOX_BLANK_OUTLINE: { family: 'ionicons', name: 'square-outline' },
  CHECKBOX_CHECKED_OUTLINE: { family: 'ionicons', name: 'checkbox-outline' },
  SAVE_OUTLINE: { family: 'ionicons', name: 'save-outline' },
  CLOSE_OUTLINE: { family: 'ionicons', name: 'close-outline' },
  CLOSE_SESSION: { family: 'ionicons', name: 'ios-exit-outline' },
  PRICETAGS_SESSION: { family: 'ionicons', name: 'pricetags-outline' }
};

export const DEFAULT_WL_PAYMENT_CARD_ICON = ICON_GLYPH_MAP.CC_VISA;

export const WLICON_GLYPH_MAP_FALLBACK = ICON_GLYPH_MAP.CLOSE;
