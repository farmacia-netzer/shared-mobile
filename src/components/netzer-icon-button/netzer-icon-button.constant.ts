export const ICONBUTTON_STYLES = {
  TRANSPARENT: 'transparent',
  FAB: 'FAB',
  OUTLINE: 'outline'
};
