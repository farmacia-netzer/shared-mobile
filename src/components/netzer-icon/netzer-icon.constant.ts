export const DEFAULT_ICON_SIZE = 22;

export interface GLYPH {
  family: string;
  name: string;
}
export interface ICON_GLYPH {
  [k: string]: GLYPH;
}
export const ICON_GLYPH_MAP: ICON_GLYPH = {
  CHEVRON_LEFT: { family: 'netzer', name: 'netzer-chevron-left' },

  ADD_SHOPPING_CART: { family: 'netzer', name: 'netzer-add-to-cart' },
  ADD_SHOPPING_LIST: { family: 'netzer', name: 'netzer-list-add' },
  ADD_CONTACT_INFO: { family: 'netzer', name: 'netzer-add-contact-info' },
  ADD_TIMESLOT: { family: 'netzer', name: 'netzer-select-timeslot' },
  AGE_RESTRICTED: { family: 'netzer', name: 'netzer-identification' },
  BARCODE_SCAN: { family: 'netzer', name: 'netzer-barcode-scan' },
  CALENDAR: { family: 'netzer', name: 'netzer-calendar' },
  CHEVRON_RIGHT: { family: 'netzer', name: 'netzer-chevron-right' },
  CLEAR: { family: 'netzer', name: 'netzer-close' },
  CLIP: { family: 'netzer', name: 'netzer-cut' },
  COUPON_CLIP: { family: 'netzer', name: 'netzer-coupon' },
  CLOSE: { family: 'netzer', name: 'netzer-close-1' },
  EDIT_PENCIL: { family: 'netzer', name: 'netzer-edit-pencil' },
  FORBIDDEN: { family: 'netzer', name: 'netzer-forbidden' },
  MENU: { family: 'netzer', name: 'netzer-hamburger' },
  MISSING_PAGE: { family: 'netzer', name: 'netzer-missing-page' },
  PIG_BANK: { family: 'netzer', name: 'netzer-piggybank' },
  LOCATION: { family: 'netzer', name: 'netzer-location-pin' },
  LOYALTY_POINTS_CARD: { family: 'netzer', name: 'netzer-loyalty-points-card' },
  REWARDS: { family: 'netzer', name: 'netzer-rewards' },
  SEARCH: { family: 'netzer', name: 'netzer-search' },
  SHOP: { family: 'netzer', name: 'netzer-shop' },
  SMARTPHONE: { family: 'netzer', name: 'netzer-smartphone' },
  SNAP: { family: 'netzer', name: 'netzer-snap' },
  SNAP_EBT: { family: 'netzer', name: 'netzer-snap-ebt' },
  SORT: { family: 'netzer', name: 'netzer-sort' },
  STAR: { family: 'netzer', name: 'netzer-star' },
  PLUS: { family: 'netzer', name: 'netzer-plus' },
  LOGIN: { family: 'netzer', name: 'netzer-avatar' },
  YOUTUBE: { family: 'netzer', name: 'netzer-youtube' },
  CHECK: { family: 'netzer', name: 'netzer-checkmark' },
  REFINE: { family: 'netzer', name: 'netzer-filter' },
  VIEW: { family: 'netzer', name: 'netzer-view' },
  DIRECTIONS: { family: 'netzer', name: 'netzer-directions' },
  STORE_FLYERS: { family: 'netzer', name: 'netzer-store-flyer' },
  STOREFRONT: { family: 'netzer', name: 'netzer-store' },
  CALL: { family: 'netzer', name: 'netzer-call' },
  TRIANGLE_DOWN: { family: 'netzer', name: 'netzer-triangle-down' },
  TIMER: { family: 'netzer', name: 'netzer-inprogress' },
  IMAGE: { family: 'materialcommunityicons', name: 'image' },
  LOCATE: { family: 'materialicons', name: 'my-location' },
  TRASH_CAN: { family: 'netzer', name: 'netzer-trash' },
  OK_CIRCLED: { family: 'netzer', name: 'netzer-ok-circled' },
  CANCEL_CIRCLED: { family: 'netzer', name: 'netzer-cancel-circled' },
  CHEVRON_UP: { family: 'netzer', name: 'netzer-chevron-up' },
  CHEVRON_THIN_UP: { family: 'netzer', name: 'netzer-thin-chevron-up' },
  ALERT_CIRCLE: { family: 'netzer', name: 'netzer-alert-circle' },
  DELIVERY_TRUCK: { family: 'netzer', name: 'netzer-delivery-truck' },
  SHOP_BAG: { family: 'netzer', name: 'netzer-shop-bag' },
  NAVIGATION: { family: 'netzer', name: 'netzer-navigation' },
  ARROW_UP_LEFT: { family: 'netzer', name: 'netzer-arrow-up-left' },
  CART_PICKUP: { family: 'netzer', name: 'netzer-pickup' },
  INFO: { family: 'materialcommunityicons', name: 'information' },
  RADIO_BOX_UNSELECTED: { family: 'materialcommunityicons', name: 'radiobox-blank' },
  RADIO_BOX_SELECTED: { family: 'materialcommunityicons', name: 'radiobox-marked' },
  ADD_PAYMENT_CARD: { family: 'netzer', name: 'netzer-add-payment-card' },
  NOTES: { family: 'netzer', name: 'netzer-notes' },
  ADD_TO_LIST: { family: 'netzer', name: 'netzer-add-to-list' },
  SHARE_IOS: { family: 'netzer', name: 'netzer-share-ios' },
  SHARE_ANDROID: { family: 'netzer', name: 'netzer-share-android' },
  DEAL_GROUP: { family: 'netzer', name: 'netzer-deal-group' },

  // ionicons
  HOME: { family: 'ionicons', name: 'home-outline' },
  PRODUCTS: { family: 'ionicons', name: 'color-filter-outline' },
  CART: { family: 'ionicons', name: 'cart' },
  PROFILE: { family: 'ionicons', name: 'person-outline' },
  ELLIPSIS_H: { family: 'ionicons', name: 'ios-ellipsis-horizontal' },
  TRASH_OUTLINE: { family: 'ionicons', name: 'trash-outline' },
  CART_OUTLINE: { family: 'ionicons', name: 'cart-outline' },
  REMOVE_OUTLINE: { family: 'ionicons', name: 'remove-outline' },
  ADD_OUTLINE: { family: 'ionicons', name: 'add-outline' },
  VERTICAL_ELLIPSIS: { family: 'ionicons', name: 'ios-ellipsis-vertical' },
  CHECKBOX_BLANK_OUTLINE: { family: 'ionicons', name: 'square-outline' },
  CHECKBOX_CHECKED_OUTLINE: { family: 'ionicons', name: 'checkbox-outline' },
  SAVE_OUTLINE: { family: 'ionicons', name: 'save-outline' },
  CLOSE_OUTLINE: { family: 'ionicons', name: 'close-outline' },
  CLOSE_SESSION: { family: 'ionicons', name: 'ios-exit-outline' },
  PRICETAGS_SESSION: { family: 'ionicons', name: 'pricetags-outline' },
  HEART_SESSION: { family: 'ionicons', name: 'heart-outline' },
  CHECK_CIRCLE_SHARP: { family: 'ionicons', name: 'checkmark-circle-sharp' },
  CHECK_OUTLINE: { family: 'ionicons', name: 'checkmark-outline' },
  TIME_OUTLINE: { family: 'ionicons', name: 'time-outline' },
  CAMERA_OUTLINE: { family: 'ionicons', name: 'camera-outline' },
  MAIL_OUTLINE: { family: 'ionicons', name: 'mail-outline' },
  LOCK_CLOSED_OUTLINE: { family: 'ionicons', name: 'lock-closed-outline' },
  PHONE_PORTRAIT_OUTLINE: { family: 'ionicons', name: 'phone-portrait-outline' },

  CASH: { family: 'ionicons', name: 'cash-outline' },
  PEOPLE: { family: 'ionicons', name: 'people-outline' },
  TROPHY: { family: 'ionicons', name: 'trophy-outline' },
  FOLDER: { family: 'ionicons', name: 'folder-open-outline' },

  CIRCLE: { family: 'ionicons', name: 'ellipse-outline' },
  COPY: { family: 'ionicons', name: 'copy-outline' },
  CIRCLE_CHECKED: { family: 'ionicons', name: 'checkmark-circle-outline' },
};

export const DEFAULT_WL_PAYMENT_CARD_ICON = ICON_GLYPH_MAP.CC_VISA;

export const WLICON_GLYPH_MAP_FALLBACK = ICON_GLYPH_MAP.CLOSE;
