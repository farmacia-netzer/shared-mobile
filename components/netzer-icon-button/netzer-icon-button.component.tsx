import React from 'react';
import { TouchableOpacity, View } from 'react-native';

import { GLYPH } from '../netzer-icon/netzer-icon.constant';
import { NetzerIcon } from '../netzer-icon/netzer-icon.component';
import { ICONBUTTON_STYLES } from './netzer-icon-button.constant';
import { styles, DEFAULT_ICON_COLOR } from './netzer-icon-button.style';

export interface NetzerIconButtonProps {
  glyph: GLYPH;
  labelKey?: any;
  testID: string;
  size?: number;
  color?: string;
  styleName?: any;
  isFAB?: boolean;
  iconStyle?: any;
  iconContainerStyle?: any;
  textStyle?: any;
  textStyleName?: string;
  buttonStyle?: any;
}

export const NetzerIconButton: React.FC<NetzerIconButtonProps> = ({
  glyph,
  testID,
  size = 20,
  color = DEFAULT_ICON_COLOR,
  styleName = ICONBUTTON_STYLES.TRANSPARENT,
  iconStyle,
  iconContainerStyle,
  buttonStyle
}: NetzerIconButtonProps) => {
  const elementStyle = {
    ...styles.container,
    ...(styleName === ICONBUTTON_STYLES.FAB && styles.FAB),
    ...buttonStyle
  };

  const elementIconStyle = {
    ...styles.defaultIconStyle,
    ...iconStyle
  };

  const activeOpacity = 0.9;
  const iconTestID = testID + '-ICON';

  const elementIconContainerStyle = {
    ...iconContainerStyle,
    ...(styleName === ICONBUTTON_STYLES.OUTLINE && styles.iconOutline)
  };

  return (
    <TouchableOpacity testID={testID} style={elementStyle} activeOpacity={activeOpacity} accessibilityRole={'button'}>
      <View style={styles.contentContainer}>
        <View style={elementIconContainerStyle}>
          <NetzerIcon testID={iconTestID} glyph={glyph} size={size} color={color} style={elementIconStyle} />
        </View>
      </View>
    </TouchableOpacity>
  );
};
