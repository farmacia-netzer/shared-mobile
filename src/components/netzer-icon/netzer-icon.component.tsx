import React, { Fragment } from 'react';
import { Animated, StyleProp, Text, ViewStyle } from 'react-native';
import { createIconSetFromFontello } from 'react-native-vector-icons';


import Feather from 'react-native-vector-icons/Feather';
import Ionicons from 'react-native-vector-icons/Ionicons';
import Foundation from 'react-native-vector-icons/Foundation';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import { DEFAULT_ICON_SIZE, GLYPH } from './netzer-icon.constant';
import { GRAY_SCALE } from '../../theme/grayscale.constant';
import netzerConfig from '../../assets/netzer-icons.json';

interface NetzerIconProps {
  glyph: GLYPH;
  size?: number;
  color?: string;
  testID?: string;
  style?: StyleProp<ViewStyle>;
  isAnimated?: boolean;
}

export const NetzerIcon: React.FC<NetzerIconProps> = ({
  glyph,
  size = DEFAULT_ICON_SIZE,
  color = GRAY_SCALE.BLACK,
  style,
  testID,
  isAnimated,
  ...props
}: NetzerIconProps) => {
  const NetzerIconConfig = createIconSetFromFontello(netzerConfig);

  const componentMap = {
    ionicons: Ionicons,
    feather: Feather,
    foundation: Foundation,
    fontawesome: FontAwesome,
    netzer: NetzerIconConfig
  };

  const Icon = componentMap[glyph.family];
  const AnimatedIcon = Animated.createAnimatedComponent(Icon);

  const iconProps = {
    size,
    color,
    style,
    testID: testID || 'NETZER-ICON-' + (glyph.name || 'home-outline').toUpperCase(),
    name: glyph.name || 'home-outline'
  };

  return (
    <Fragment>
      {isAnimated ? (
        <AnimatedIcon {...iconProps} {...props} />
      ) : (
        <Text {...props}>
          <Icon {...iconProps} />
        </Text>
      )}
    </Fragment>
  );
};
