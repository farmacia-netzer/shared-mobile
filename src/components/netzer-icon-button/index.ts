import { styles, FAB_DIMENSIONS } from './netzer-icon-button.style';

export * from './netzer-icon-button.component';
export const ICON_BUTTON_STYLES = styles;
export { FAB_DIMENSIONS };
