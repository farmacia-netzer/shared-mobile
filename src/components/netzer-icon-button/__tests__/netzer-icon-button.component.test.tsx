import React from 'react';
import { render } from '@testing-library/react-native';

import { ICON_GLYPH_MAP } from '##component/netzer-icon/netzer-icon.constant';

import { NetzerIconButton } from '../netzer-icon-button.component';

describe('<NetzerIconButton />', () => {
  it('renders', () => {
    const { toJSON } = render(
      <NetzerIconButton glyph={ICON_GLYPH_MAP.HOME} testID={'WLICON-BUTTON'} labelKey={'WLICON-BUTTON'} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  // it('renders without caption', () => {
  //   const { queryByTestId } = render(
  //     <WLIconButton glyph={WLICON_GLYPH_MAP.MENU} testID={'WLICON-BUTTON'} labelKey={'WLICON-BUTTON'} />
  //   );
  //   const element = queryByTestId('WLICON-BUTTON');
  //   expect(element).not.toBeNull();
  //   const iconElement = queryByTestId('WLICON-BUTTON-ICON');
  //   expect(iconElement).not.toBeNull();
  //   const textElement = queryByTestId('WLICON-BUTTON-TEXT');
  //   expect(textElement).toBeNull();
  // });

  // it('renders with caption', () => {
  //   const { queryByTestId } = render(
  //     <WLIconButton
  //       glyph={WLICON_GLYPH_MAP.MENU}
  //       testID={'WLICON-BUTTON'}
  //       labelKey={'WLICON-BUTTON'}
  //       isCaptioned={true}
  //     />
  //   );
  //   const element = queryByTestId('WLICON-BUTTON');
  //   expect(element).not.toBeNull();
  //   const iconElement = queryByTestId('WLICON-BUTTON-ICON');
  //   expect(iconElement).not.toBeNull();
  //   const textElement = queryByTestId('WLICON-BUTTON-TEXT');
  //   expect(textElement).not.toBeNull();
  // });

  // it('renders additional props as provided', () => {
  //   const { getByTestId } = render(
  //     <WLIconButton
  //       glyph={WLICON_GLYPH_MAP.MENU}
  //       testID={'WLICON-BUTTON'}
  //       labelKey={'WLICON-BUTTON'}
  //       onPress={() => {}}
  //     />
  //   );
  //   const element = getByTestId('WLICON-BUTTON');
  //   expect(element.props.onPress).toBeTruthy();
  // });

  // it('overrides styles when given', () => {
  //   const { getByTestId } = render(
  //     <WLIconButton
  //       glyph={WLICON_GLYPH_MAP.MENU}
  //       testID={'WLICON-BUTTON'}
  //       labelKey={'WLICON-BUTTON'}
  //       textStyle={{ color: 'red' }}
  //       iconStyle={{ color: 'blue' }}
  //       buttonStyle={{ backgroundColor: 'green' }}
  //       isCaptioned={true}
  //     />
  //   );
  //   const element = getByTestId('WLICON-BUTTON');
  //   expect(element.props.style.backgroundColor).toBe('green');
  //   const WLICON = getByTestId('WLICON-BUTTON-ICON').props.children.props;
  //   expect(WLICON.style.color).toBe('blue');
  //   const WLTEXT = getByTestId('WLICON-BUTTON-TEXT').props;
  //   expect(WLTEXT.style.color).toBe('red');
  // });

  // it('overrides props when given', () => {
  //   const { getByTestId } = render(
  //     <WLIconButton
  //       glyph={WLICON_GLYPH_MAP.MENU}
  //       testID={'WLICON-BUTTON'}
  //       labelKey={'WLICON-BUTTON'}
  //       accessibilityLabel={'ABC'}
  //       accessible={false}
  //     />
  //   );
  //   const element = getByTestId('WLICON-BUTTON');
  //   expect(element.props.accessible).toBe(false);
  //   expect(element.props.accessibilityLabel).toBe('ABC');
  // });

  // it('appends FAB styles when isFAB is true', () => {
  //   const { getByTestId } = render(
  //     <WLIconButton
  //       glyph={WLICON_GLYPH_MAP.MENU}
  //       testID={'WLICON-BUTTON'}
  //       labelKey={'WLICON-BUTTON'}
  //       styleName={'FAB'}
  //     />
  //   );
  //   const element = getByTestId('WLICON-BUTTON');
  //   const buttonWidth = element.props.style.width;
  //   expect(element.props.style.borderRadius).toBe(buttonWidth / 2);
  // });

  // test('default FAB style passes Accessbility Color Test', () => {
  //   const { getByTestId } = render(
  //     <WLIconButton
  //       glyph={WLICON_GLYPH_MAP.MENU}
  //       testID={'WLICON-BUTTON'}
  //       labelKey={'WLICON-BUTTON'}
  //       styleName={'FAB'}
  //     />
  //   );
  //   const element = getByTestId('WLICON-BUTTON');
  //   const backgroundColor = element.props.style.backgroundColor;
  //   const WLICON = getByTestId('WLICON-BUTTON-ICON').props.children.props;
  //   isA11yColorTestPassed(backgroundColor, WLICON.color, WLICON.size);
  // });

  // it('has default accessibility props and uses labelKey for accessibility label', () => {
  //   const { getByTestId } = render(
  //     <WLIconButton glyph={WLICON_GLYPH_MAP.MENU} testID={'WLICON-BUTTON'} labelKey={'WLICON-BUTTON'} />
  //   );
  //   const element = getByTestId('WLICON-BUTTON');
  //   expect(element.props.accessible).toBe(true);
  //   expect(element.props.accessibilityLabel).toBe('WLICON-BUTTON');
  //   expect(element.props.accessibilityRole).toBe('button');
  // });

  // it('renders fallback label when labelKey is an empty string', () => {
  //   const { getByTestId } = render(
  //     <WLIconButton glyph={WLICON_GLYPH_MAP.MENU} testID={'WLICON-BUTTON'} labelKey={''} />
  //   );
  //   const element = getByTestId('WLICON-BUTTON');
  //   expect(element.props.accessibilityLabel).toBe('WLICONBUTTON_LABEL');
  // });

  // it('renders correct accessibilityLabel value when given', () => {
  //   const { getByTestId } = render(
  //     <WLIconButton glyph={WLICON_GLYPH_MAP.MENU} testID={'WLICON-BUTTON'} labelKey={''} accessibilityLabel={'ABC'} />
  //   );
  //   const element = getByTestId('WLICON-BUTTON');
  //   expect(element.props.accessibilityLabel).toBe('ABC');
  // });
});
