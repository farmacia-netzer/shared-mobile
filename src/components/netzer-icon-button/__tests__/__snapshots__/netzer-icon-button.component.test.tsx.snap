// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NetzerIconButton /> renders 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="WLICON-BUTTON"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontSize": 20,
              },
              Object {
                "justifyContent": "center",
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="WLICON-BUTTON-ICON"
        >
          Ôçá
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
